integration/tests/access/cohort4/websocket_subscriptions_test.go

```diff
@@ -0,0 +1,1201 @@
+			subscriptionRequest := s.subscribeMessageRequest(
+				"dummy_id",
+				tt.topic,
+				tt.prepareArguments(),
+			)
+			s.Require().NoError(wsClient.WriteJSON(subscriptionRequest))
+
+			// Step 3: Listen for WebSocket responses for the specified duration
+			responses, baseMessageResponses, _ := s.listenWebSocketResponses(
+				wsClient,
+				tt.listenSubscriptionResponseDuration,
+				subscriptionRequest.SubscriptionID,
+			)
+
+			// Step 4: Validate the subscription response
+			s.Require().Equal(1, len(baseMessageResponses), "expected one subscription response")
+			s.validateBaseMessageResponse(baseMessageResponses[0])
+
+			// Step 5: Use the provided validation function to check received responses
+			s.validate(
+				subscriptionRequest.SubscriptionID,
+				subscriptionRequest.Topic,
+				responses,
+			)
+
+			// Step 6: Optionally unsubscribe from the topic
+			if tt.testUnsubscribe {
+				// Create an unsubscription request
+				unsubscriptionRequest := s.unsubscribeMessageRequest(subscriptionRequest.SubscriptionID)
+
+				// Send the unsubscription request to the WebSocket server
+				s.Require().NoError(wsClient.WriteJSON(unsubscriptionRequest))
+
+				// Step 6.1: Read and validate the unsubscription response
+				var response models.BaseMessageResponse
+				err := wsClient.ReadJSON(&response)
+				s.Require().NoError(err, "failed to read unsubscription response")
+				s.validateBaseMessageResponse(response)
+			}
+		})
+	}
+}
+
+// testSubscriptionMultiplexing verifies that when subscribing to multiple channels,
+// all expected messages are received correctly, ensuring subscription multiplexing works as expected.
+func (s *WebsocketSubscriptionSuite) testSubscriptionMultiplexing() {
+	subscriptions := []struct {
+		name             string
+		topic            string
+		prepareArguments func() models.Arguments
+	}{
+		{
+			name:  "Blocks streaming",
+			topic: data_providers.BlocksTopic,
+			prepareArguments: func() models.Arguments {
+				return models.Arguments{"block_status": parser.Finalized}
+			},
+		},
+		{
+			name:  "Block headers streaming",
+			topic: data_providers.BlockHeadersTopic,
+			prepareArguments: func() models.Arguments {
+				return models.Arguments{"block_status": parser.Finalized}
+			},
+		},
+		{
+			name:  "Block digests streaming",
+			topic: data_providers.BlockDigestsTopic,
+			prepareArguments: func() models.Arguments {
+				return models.Arguments{"block_status": parser.Finalized}
+			},
+		},
+	}
+
+	// Step 1: Establish a WebSocket connection
+	wsClient, err := common.GetWSClient(s.ctx, getWebsocketsUrl(s.restAccessAddress))
+	s.Require().NoError(err)
+	defer func() { s.Require().NoError(wsClient.Close()) }()
+
+	responses := make([]models.BaseDataProvidersResponse, len(subscriptions))
+	// Step 2: Create and send subscription requests for all topics
+	subscriptionRequests := make(map[string]models.SubscribeMessageRequest)
+	for i, sub := range subscriptions {
```

Comment:
this isn't really testing multiplexing, but instead subscribing to one topic, then listening for responses, then subscribing to the next, etc.

Instead, I think this test should behave more like an application:
1. Setup the router that collects messages into some buckets
2. Open subscriptions for all data it wants
3. After some period of time unsubscribe from all channels
4. Review all of the data to make sure it makes sense.
	* Got subscribe and unsubscribe responses
	* Got messages for consistent blocks across all subscriptions